src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -183,6 +189,34 @@ public TracerSettings(IConfigurationSource source)
         /// <seealso cref="ConfigurationKeys.AgentPort"/>
         public Uri AgentUri { get; set; }
 
+        /// <summary>
+        /// Gets or sets the key used to determine the transport for sending traces.
+        /// Default is null.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TracesTransport"/>
+        public string TracesTransport { get; set; }
+
+        /// <summary>
+        /// Gets or sets the windows pipe name where the Tracer can connect to the Agent.
+        /// Default is null.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TracesPipeName"/>
+        public string TracesPipeName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the timeout in milliseconds for the windows named pipe requests.
+        /// Default is 0.
```

Comment:
Does `0` mean there is no timeout? If so, we should document that here, and we should consider making this `int?`. I think `null` expresses "no timeout" better than `0` does.

Reply:
In .NET, the convention for infinite timeout is -1.
Also enforced by the `Timeout.Infinite` constant: https://docs.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinite?view=net-5.0